apiserver/apiserver.go

```diff
@@ -697,6 +701,11 @@ func (srv *Server) endpoints() ([]apihttp.Endpoint, error) {
 					Handler: h,
 					Query:   ":bucket",
 				}
+			} else if strings.HasPrefix(handler.pattern, objectsRoutePrefix) {
```

Comment:
can we replace all these if/else with a `switch` statement? It could quickly become unreadable

Reply:
The whole thing is a mess. If I touch it too much, I'm going to re-write it.